packages/server/modules/shared/helpers/dbHelper.ts

```diff
@@ -107,17 +107,20 @@ export const numberOfFreeConnections = (knex: Knex) => {
 }
 
 export const withTransaction = async <T>(
-  callback: Promise<T> | T,
-  trx: Knex.Transaction
-): Promise<T> => {
+  operation: (args: { db: Knex }) => MaybeAsync<T>,
```

Comment:
the old implementation requires you to manually create the trx beforehand which makes it less useful than it could be. the trx object should be an internal implementation detail, you don't need it outside of the function.

i also made the API more similar to withOperationLogging, with the goal of possibly merging withTransaction+withOperationLogging+commandFactory together into 1 utility in the future